File: euler/hs/019.hs
import Data.Time

isSunday :: Day -> Bool
isSunday d = (diffDays d (fromGregorian 1900 1 1) + 1) `mod` 7 == 0

firstDaysOfMonth :: Integer -> Integer -> [Day]
firstDaysOfMonth from to = [fromGregorian y m 1 | m <- [1..12], y <- [from..to]]

main :: IO()
main = print $ length $ filter isSunday (firstDaysOfMonth 1901 2000)

File: euler/py/019.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from datetime import date

def is_sunday(d):
    return ((d - date(1900, 1, 1)).days + 1) % 7 == 0


def first_days_of_month(year_from, year_to):
    for y in xrange(year_from, year_to + 1):
        for m in xrange(1, 13):
            yield date(y, m ,1)

if __name__ == '__main__':
    print len(filter(is_sunday, first_days_of_month(1901, 2000)))
